src/libexpr/eval.cc

```diff
@@ -2423,10 +2445,15 @@ SourcePath EvalState::coerceToPath(const PosIdx pos, Value & v, NixStringContext
     if (v.type() == nPath)
         return v.path();
 
-    /* Similarly, handle __toString where the result may be a path
+    /* Similarly, handle outPath and __toString where the result may be a path
        value. */
     if (v.type() == nAttrs) {
-        auto i = v.attrs->find(sToString);
+        auto i = v.attrs->find(sOutPath);
+        if (i != v.attrs->end()) {
+            return coerceToPath(pos, *i->value, context, errorCtx);
+        }
```

Comment:
This rule is not new. Previously, it would have worked by falling through to the `coerceToString` + `rootPath` code down below.